internal/receiver/smartagentreceiver/converter/traces.go

```diff
@@ -24,12 +25,51 @@ import (
 	"go.uber.org/zap"
 )
 
+const (
+	sfxDataSourceKey         = "data_source_ip"
+	resourceClientIPAttrName = "ip"
+)
+
 var zipkinv2Translator = zipkinv2.ToTranslator{ParseStringTags: false}
 
 func sfxSpansToPDataTraces(spans []*trace.Span, logger *zap.Logger) (pdata.Traces, error) {
-	// SFx trace is effectively zipkin, so more convenient to convert to it and then rely
-	// on existing zipkin receiver translator
-	return zipkinv2Translator.ToTraces(sfxToZipkinSpans(spans, logger))
+	// we batch sfx spans by the client IP that reported the spans to collector. Each
+	// batch gets translated separately to ensure that spans from sources with different
+	// IPs don't get bundled together under the same resource.
+	batches := map[string][]*trace.Span{}
+	for _, span := range spans {
+		if span == nil {
+			continue
+		}
+		var sourceIP string
+		if val, ok := span.Meta[sfxDataSourceKey]; ok {
+			if ip, ok := val.(net.IP); ok {
+				sourceIP = ip.String()
+			}
```

Comment:
Should we add couple of debug logs here when sfxDataSourceKey is missing and it doesn't contain an IP?

Reply:
Not sure if worth it. logger instance is not available here so probably need to change a few places. I think the it would be obvious that IP was not detected so not sure how much it'd help but can add if you feel strongly about it.

Reply:
I believe the `logger` here is the receiver's logger passed through the output's converter.

Reply:
Empty string doesn't break this functionality, but k8s_tagger won't be able to do its job. Chatty logs on debug level should be fine I guess. Having debug logs makes it easier to find out why correlation doesn't work next time. But i'm ok with keeping it as is if adding logs is not straight-forward.